gc/base/MemoryPoolAddressOrderedList.hpp

```diff
@@ -75,9 +82,17 @@ class MM_MemoryPoolAddressOrderedList : public MM_MemoryPoolAddressOrderedListBa
 	void updateHintsBeyondEntry(MM_HeapLinkedFreeHeader *freeEntry);
 	void *internalAllocate(MM_EnvironmentBase *env, uintptr_t sizeInBytesRequired, bool lockingRequired, MM_LargeObjectAllocateStats *largeObjectAllocateStats);
 	bool internalAllocateTLH(MM_EnvironmentBase *env, uintptr_t maximumSizeInBytesRequired, void * &addrBase, void * &addrTop, bool lockingRequired, MM_LargeObjectAllocateStats *largeObjectAllocateStats);
+	MMINLINE bool doesNeedAlignment(MM_EnvironmentBase *env, MM_HeapLinkedFreeHeader *freeEntry)
+	{
+		bool ret = false;
+		if ((uintptr_t) freeEntry >= (uintptr_t)_firstUnAlignmentFreeEntry) {
+			ret = true;
+		}
+		return ret;
```

Comment:
I think you can just do it in one line, and need no casting (comparison of pointers to same types is valid):

return (freeEntry >= _firstUnalignedFreeEntry);

Reply:
changed